include/boost/gil/image_processing/morphology.hpp

```diff
@@ -0,0 +1,131 @@
+
+    for(std::ptrdiff_t struct_elem_y = 0; struct_elem_y < struct_elem_height; ++struct_elem_y)
+    {
+        // The cooresponding value to be read from src_view is calculated as
+        // (src_y + struct_elem_y - struct_elem_height/2, src_x + struct_elem_x - struct_elem_width/2)
+        // NOTE: This calculation is only valid when the struct_elem dimensions are odd.
+        auto y_to_read = src_y + struct_elem_y - struct_elem_height/2;
+        if(y_to_read < 0 || y_to_read >= src_view.height())
+            continue;
+        typename SrcView::x_iterator src_it = src_view.row_begin(y_to_read);
+
+        for(std::ptrdiff_t struct_elem_x = 0; struct_elem_x < struct_elem_width; ++struct_elem_x)
+        {
+            // We need to consider only those positions which have 1 in Structuring Element
+            if(struct_elem[struct_elem_x][struct_elem_y] == 1)
+            {
+                auto x_to_read = src_x + struct_elem_x - struct_elem_width/2;
+                if(x_to_read < 0 || x_to_read >= src_view.width())
+                    continue;
+                auto src_point_val = src_it[x_to_read];
+                if (src_point_val < min)
+                    min = src_point_val;
+                if (src_point_val > max)
+                    max = src_point_val;
+            }
+        }
+    }
+
+    typename DstView::x_iterator dst_it = dst_view.row_begin(src_y);
+
+    if (op == morphology_operator::erode)
+        dst_it[src_x] = min;
+    else if(op == morphology_operator::dilate)
+        dst_it[src_x] = max;
+}
+
+template <typename SrcView, typename DstView, typename StructElement>
+void morph_impl(SrcView const &src_view, DstView &dst_view, StructElement const &struct_elem, morphology_operator const &op)
+{
+    BOOST_ASSERT(src_view.dimensions() == dst_view.dimensions());
+    BOOST_ASSERT(struct_elem.size() != 0 && struct_elem[0].size() != 0);
+
+    std::size_t height = src_view.height();
+    std::size_t width = dst_view.width();
+
+    for(std::ptrdiff_t src_y = 0; src_y < height; ++src_y)
+    {
+        for(std::ptrdiff_t src_x = 0; src_x < width; ++src_x)
+        {
+            convolve_with_struct_element<SrcView, DstView, StructElement>(src_view, dst_view, struct_elem, src_y, src_x, op);
+        }
+    }
+}
+
+} //namespace boost::gil::detail
+
+/// \addtogroup ImageProcessing
+/// @{
+
+/// \brief Performs Morphological Dilation of the given image with a given Structuring Element.
+/// \param src_view - The source image view.
+/// \param dst_view - The destination image view (after performing dilation).
+/// \param struct_elem - The Structuing Element. Should be a 2D-array type consiting of binary int values (0 and 1).
+/// \tparam SrcView type of source image. Should confirm to gil::ImgaeViewConcept.
+/// \tparam DstView type of destination image. Should confirm to gil::MutableImageViewConcept.
+/// \tparam StructElement type of structuring element. Should provide size() and operator[].
```

Comment:
In future, for such kernel-like containers, we may have `KernelConcept` or similar which partially satisfies the requirements of [SequenceContainer](https://en.cppreference.com/w/cpp/named_req/SequenceContainer).

Reply:
Will implement it during the GSoC period :)

Reply:
Sure. I added "Further development during GSoC 2020" section to the PR description.